packages/block-library/src/button/style.scss

```diff
@@ -9,6 +9,8 @@ $blocks-block__margin: 0.5em;
 	text-align: center;
 	word-break: break-word; // overflow-wrap doesn't work well if a link is wrapped in the div, so use word-break here.
 	box-sizing: border-box;
+	// Stretch to button width if it is set.
+	width: 100%;
```

Comment:
This is a bit of a nitpick, but I'm wondering if it'd be safer to move this to the block where we have rules for `.wp-block-buttons > .wp-block-button` so that it only applies to button blocks that are the direct child of the core Buttons block? From memory, there can be some difficult cases either with super old markup from back when the Button block didn't need to be in a Buttons container, or with plugins that borrow / extend styles from the Button block, so I was wondering if that might help restrict the scope of where this gets applied 🤔... of course I very well might be overthinking this!

Reply:
I can't think of a situation where we _wouldn't_ want the interactive element to occupy the full width of its wrapper, because having something styled as a button not being clickable in its whole area would be really crappy UX. But I'm not really sure why Button needed a wrapper in the first place so 🤷 happy to leave it alone for now. 

I'd prefer to make a direct change like this instead of adding all sorts of specificity to it, as it seems like a sensible default, but maybe better do it earlier in the release cycle.

Reply:
> I'd prefer to make a direct change like this instead of adding all sorts of specificity to it, as it seems like a sensible default, but maybe better do it earlier in the release cycle.

Totally agree, I think it sounds like a great default and I also prefer avoiding the extra specificity where we can. I'm mostly just cautious because of having previously merged block CSS changes I thought were safe and then found out later that I'd broken something edge-casey.

Since it'd arguably be fixing a bug, once this PR lands, I suppose we could open up a separate PR just for the Button CSS change and then ping for wider review from designers, etc? Since it'd be a bug fix we might have a chance of still landing it for 6.2 past the cutoff